### ISSUE TITLE
Continuous Messages of exporting failed, dropped data, sender failed in the aws-otel-collector.log

### ISSUE THREAD

**Issue by georges-git** [aws-observability/aws-otel-collector#551]
Hello @mxiamxia and AWS group -  Keep getting following messages in the OTEL collector log. How to fix it?

{2021-06-23 09:52:04.43621453 -0400 EDT m=+120.147512184, Level:error, Caller:go.opentelemetry.io/collector@v0.27.0/exporter/exporterhelper/queued_retry.go:173, Message:Exporting failed. Dropping data. Try enabling sending_queue to survive temporary failures., Stack:go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).send
	go.opentelemetry.io/collector@v0.27.0/exporter/exporterhelper/queued_retry.go:173
go.opentelemetry.io/collector/exporter/exporterhelper.NewMetricsExporter.func2
	go.opentelemetry.io/collector@v0.27.0/exporter/exporterhelper/metrics.go:103
go.opentelemetry.io/collector/consumer/consumerhelper.ConsumeMetricsFunc.ConsumeMetrics
	go.opentelemetry.io/collector@v0.27.0/consumer/consumerhelper/metrics.go:29
go.opentelemetry.io/collector/service/internal/fanoutconsumer.metricsConsumer.ConsumeMetrics
	go.opentelemetry.io/collector@v0.27.0/service/internal/fanoutconsumer/consumer.go:51
go.opentelemetry.io/collector/processor/batchprocessor.(*batchMetrics).export
	go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:285
go.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).sendItems
	go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:183
go.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).startProcessingCycle
	go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:144}
{2021-06-23 09:52:04.436239353 -0400 EDT m=+120.147537026, Level:warn, Caller:go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:184, Message:Sender failed, Stack:}

**Comment by mxiamxia** [comment]
could you attach more collector logs?

**Comment by georges-git** [comment]
Hello @mxiamxia - Let me what more logs you are looking for. Keep getting errors:
[opentelemetry.auto.trace 2021-06-24 09:12:43:741 -0400] [grpc-default-executor-58] WARN io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter - Failed to export spans. Error message: UNAVAILABLE: io exception
[opentelemetry.auto.trace 2021-06-24 09:12:43:741 -0400] [grpc-default-executor-58] WARN io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter - Failed to export spans. Error message: UNAVAILABLE: io exception

**Comment by georges-git** [comment]
Hello @mxiamxia - This is all the information in the collector logs. Let me know how to fix this.
 
go.opentelemetry.io/collector/consumer/consumerhelper.ConsumeMetricsFunc.ConsumeMetrics
                go.opentelemetry.io/collector@v0.27.0/consumer/consumerhelper/metrics.go:29
go.opentelemetry.io/collector/processor/batchprocessor.(*batchMetrics).export
                go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:285
go.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).sendItems
                go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:183
go.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).startProcessingCycle
                go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:144}
{2021-06-25 17:47:42.118233801 -0400 EDT m=+18856.163846182, Level:error, Caller:go.opentelemetry.io/collector@v0.27.0/exporter/exporterhelper/queued_retry.go:173, Message:Exporting failed. Dropping data. Try enabling sending_queue to survive temporary failures., Stack:go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).send
                go.opentelemetry.io/collector@v0.27.0/exporter/exporterhelper/queued_retry.go:173
go.opentelemetry.io/collector/exporter/exporterhelper.NewMetricsExporter.func2
                go.opentelemetry.io/collector@v0.27.0/exporter/exporterhelper/metrics.go:103
go.opentelemetry.io/collector/consumer/consumerhelper.ConsumeMetricsFunc.ConsumeMetrics
                go.opentelemetry.io/collector@v0.27.0/consumer/consumerhelper/metrics.go:29
go.opentelemetry.io/collector/processor/batchprocessor.(*batchMetrics).export
                go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:285
go.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).sendItems
                go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:183
go.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).startProcessingCycle
                go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:144}
{2021-06-25 17:47:42.118273569 -0400 EDT m=+18856.163885957, Level:warn, Caller:go.opentelemetry.io/collector@v0.27.0/processor/batchprocessor/batch_processor.go:184, Message:Sender failed, Stack:}

**Comment by sethAmazon** [comment]
Can you turn on debug logs please. echo "loggingLevel=DEBUG" | sudo tee -a /opt/aws/aws-otel-collector/etc/extracfg.txt then restart the collector if running on ec2

**Comment by georges-git** [comment]
Attaching again the zipped log file with DEBUG logging level. Could you escalate and help resolve this as it is being going on for last few weeks.
[aws-otel-collector.zip](https://github.com/aws-observability/aws-otel-collector/files/6743004/aws-otel-collector.zip)

**Comment by sethAmazon** [comment]
{2021-06-28 12:12:53.950912958 -0400 EDT m=+60.073175088, Level:debug, Caller:github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter@v0.22.0/cwlog_client.go:158, Message:cwlog_client: creating stream fail, Stack:}
{2021-06-28 12:12:53.950976095 -0400 EDT m=+60.073238214, Level:debug, Caller:github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter@v0.22.0/cwlog_client.go:176, Message:CreateLogStream / CreateLogGroup has errors., Stack:}
{2021-06-28 12:12:53.950990885 -0400 EDT m=+60.073252994, Level:warn, Caller:github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter@v0.22.0/pusher.go:280, Message:Failed to create stream token, Stack:github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter.(*pusher).pushLogEventBatch
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter@v0.22.0/pusher.go:280

I saw a similar error when I ran on ec2. I think there might be a problem with the way we are getting creds. Can you try running this in a docker container and passing in the access key manually to the container with "-e AWS_ACCESS_KEY_ID=<> -e AWS_SECRET_ACCESS_KEY=<>" guide on how to run with docker is https://github.com/aws-observability/aws-otel-collector/blob/main/docs/developers/docker-demo.md

**Comment by georges-git** [comment]
Hello @sethAmazon - Can you expand on what your are saying above?  My app is running in a docker. Are you asking to run the OTEL daemon process in a docker container?

**Comment by sethAmazon** [comment]
How are you passing in the credentials for otel?

**Comment by georges-git** [comment]
I am not passing any credentials. I just run - sudo /opt/aws/aws-otel-collector/bin/aws-otel-collector-ctl -c /opt/aws/aws-otel-collector/etc/config.yaml -a start.

How would you like me to pass the credentials to this "aws-otel-collector" process running on the linux hosts.

**Comment by sethAmazon** [comment]
Hi @georges-git can we do a live debug session sometime on this?

**Comment by aureliomarcoag** [comment]
I had this problem on a Python Lambda function. After tweaking the code, I could reproduce the problem by simply using `boto3.client("s3").download_file()` inside a tempfile context manager, so something like this:

```
s3_client = boto3.client("s3")
with tempfile.NamedTemporaryFile() as tmp_file:
    boto3.client("s3").download_file(Bucket="mybucket", Key="key/object", Filename=tmp_file.name)
```

I eventually stumbled upon https://github.com/aws-observability/aws-otel-lambda/issues/10 so I assume this might be related. I switched from download_file to get_object and the error went away. I also tested with upload_file and that caused the same error to happen.
